src/services/pokemon/pokemon.ts

```diff
@@ -11,6 +13,18 @@ export const pokemonApi = createApi({
     }),
     getAllPokemon: builder.query<Pokemons, void>({
       query: () => 'pokemon/',
+
+      async onQueryStarted(arg, { dispatch, queryFulfilled }) {
+        const { data } = await queryFulfilled;
+        dispatch(setPokemonsCount(data.count));
```

Comment:
I think it needs to be inside if condition. What if request failed and there is no data.count ?